Resources/Private/Fusion/NodeTypes/LoginForm.fusion

```diff
@@ -1,16 +1,125 @@
 ##
-# "LoginForm" element, extending "Plugin"
+# "LoginForm" element, extending "ContentComponent"
 #
-prototype(Flowpack.Neos.FrontendLogin:LoginForm) < prototype(Neos.Neos:Plugin) {
-	package = 'Flowpack.Neos.FrontendLogin'
-	controller = 'Authentication'
-	action = 'index'
-
-	redirectAfterLoginUri = Neos.Neos:NodeUri {
-		node = ${q(node).property('redirectAfterLogin')}
-	}
-
-	redirectAfterLogoutUri = Neos.Neos:NodeUri {
-		node = ${q(node).property('redirectAfterLogout')}
-	}
-}
\ No newline at end of file
+prototype(Flowpack.Neos.FrontendLogin:LoginForm) < prototype(Neos.Neos:ContentComponent) {
+
+    loginPrototype = 'Flowpack.Neos.FrontendLogin:LoginForm.Login'
+    logoutPrototype = 'Flowpack.Neos.FrontendLogin:LoginForm.Logout'
+
+    renderer = Neos.Fusion:Case {
+        loggedIn {
+            condition = ${Security.hasRole('Flowpack.Neos.FrontendLogin:User')}
+            type = ${props.logoutPrototype}
+            element {
+                accountIdentifier = ${Security.getAccount().accountIdentifier}
+                redirectAfterLogoutUri = Neos.Neos:NodeUri {
+                    node = ${q(node).property('redirectAfterLogout')}
+                }
+            }
+        }
+
+        default {
+            condition = true
+            type = ${props.loginPrototype}
+            element {
+                redirectAfterLoginUri = Neos.Neos:NodeUri {
+                    node = ${q(node).property('redirectAfterLogin')}
```

Comment:
This will not work when the target page needs the front-end user privilege.
In this case a url to the login page is created instead.

So I think we have to access the node in the authentication controller somehow.

Reply:
Hmm, maybe getting rid of the plugin has some side effects :-/ not sure how to solve this safe and sane.

Reply:
A way around would be to drop the redirectAfter-feature and just always send the user back to the referer wich uses arguments that are signed with an hmac.

However finding a way to transport this safely to the controller without relying on plugins would be a really nice example for such cases.

Reply:
I wonder a bit how it worked before. Did that work before?

Other ways:

* Providing a linkeditor for both uris would of course also make it possible to give a relative url. Not the best solution but maybe would anyway make sense for other usecases.
* Give the login form node identifier to the controller and let it build the uri from its properties when it has the required privileges. Would require us to duplicate some code from the NodeUri builder I guess.

Reply:
The plugin evaluates the path redirectAfterLoginUri after the login and not immediately. Maybe we could create a lazyEvaluationProxy fusionObject as wrapper for the NodeUri :-/

BTW: I just found out that the FusionTemplate does similar things for all untyped properties.

Reply:
Core of the problem is that Props are usually evaluated immediately and not once they are accessed. Changing that is not that simple as you can see here https://github.com/neos/neos-development-collection/pull/2738

Reply:
So which way should we go then. Back to the plugin for now?

Reply:
Either use your original pr or find a solution here which will take time and a sprint would be a good idea.

Reply:
But then I would prefer the plugin approach for now. Would love to have this and not wait 5-6 months.
With some more experience from this we also have a better idea until the sprint to remove the plugin part.

Do you want to create another PR with my commit + your cleanup adjustments?